tests/python_tests/test_chat_generate_api.py

```diff
@@ -0,0 +1,179 @@
+        answer_str = tokenizer.decode(answer[0, tokenized['input_ids'].numel():], skip_special_tokens=True)
+        chat_history_hf.append({'role': 'assistant', 'content': answer_str})
+
+        answer_ov = pipe.generate(prompt, **generation_config)
+        chat_history_ov.append({'role': 'assistant', 'content': answer_ov})
+
+    pipe.finish_chat()
+    
+    if chat_history_ov != chat_history_hf:
+        print(f'hf_output: {chat_history_hf}')
+        print(f'ov_output: {chat_history_ov}')
+    assert chat_history_ov == chat_history_hf
+
+
+@pytest.mark.parametrize("generation_config", configs)
+@pytest.mark.parametrize("model_descr", get_chat_models_list())
+@pytest.mark.precommit
+def test_chat_compare_text_history_with_HF(model_descr, generation_config: Dict):
+    # compares with HF when history in ov_genai is save as a text
+    device = 'CPU'
+    chat_history_hf = []
+    chat_history_ov = []
+    chat_prompt = ''
+    model_id, path, tokenizer, model_opt, pipe = read_model(model_descr)
+
+    # HF in chat scenario does not add special tokens, but openvino tokenizer by default is converted with add_special_tokens=True.
+    # Since To match need to regenerate openvino_tokenizer/detokenizer.
+    ov_tokenizer, ov_detokenizer = openvino_tokenizers.convert_tokenizer(tokenizer, add_special_tokens=False, with_detokenizer=True)
+    openvino.save_model(ov_tokenizer, path / "openvino_tokenizer.xml")
+    openvino.save_model(ov_detokenizer, path / "openvino_detokenizer.xml")
+    pipe = ov_genai.LLMPipeline(str(path), device, config={"ENABLE_MMAP": False})
+    
+    for prompt in quenstions:
+        chat_history_hf.append({'role': 'user', 'content': prompt})
+        chat_history_ov.append({'role': 'user', 'content': prompt})
+        
+        chat_prompt = tokenizer.apply_chat_template(chat_history_hf, tokenize=False, add_generation_prompt=True)
+        tokenized = tokenizer(chat_prompt, return_tensors='pt', add_special_tokens=False)
+        
+        answer = model_opt.generate(**tokenized, **generation_config, do_sample=False, repetition_penalty = None)
+        answer_str = tokenizer.decode(answer[0, tokenized['input_ids'].numel():], skip_special_tokens=True)
+        chat_history_hf.append({'role': 'assistant', 'content': answer_str})
+        
+        chat_prompt = pipe.get_tokenizer().apply_chat_template(chat_history_ov, add_generation_prompt=True)
+        answer_ov = pipe.generate(chat_prompt, **generation_config)
+        chat_history_ov.append({'role': 'assistant', 'content': answer_ov})
+  
+    if chat_history_ov != chat_history_hf:
+        print(f'hf_output: {chat_history_hf}')
+        print(f'ov_output: {chat_history_ov}')
+    assert chat_history_ov == chat_history_hf
+
+
+@pytest.mark.parametrize("generation_config", configs)
+@pytest.mark.parametrize("model_descr", get_chat_models_list())
+@pytest.mark.precommit
+def test_chat_compare_statefull_vs_text_history(model_descr, generation_config: Dict):
```

Comment:
This seems to be redundant: if `chat_history_ov == chat_history_hf with add_generation_prompt=False` and `chat_history_ov == chat_history_hf with add_generation_prompt=True` then `chat_history_ov with add_generation_prompt=False` == `chat_history_ov with add_generation_prompt=False`.

Reply:
There are some tests where HF does not match to genai in both `test_chat_compare_text_history_with_HF` `test_chat_compare_with_HF`. This can happen either generate was wrong, either we did something wrong with keeping/restoring history.

To ensure that we correctly kept history in KC cache and continued i added this test as well, it still can be greed while others are red.